### PR TITLE
Call default state for zwave-siren capabilities

### PR DESCRIPTION
**Pull request by ctowns**
https://smartthings.atlassian.net/browse/CHAD-9325

### CODE REVIEW

**Comment by ctowns** [comment]
Any suggestions on how I should go about testing this? The unit tests are still passing but I am unsure how to check that this addresses the missing state behavior in described in the ticket.

File: drivers/SmartThings/zwave-siren/src/init.lua [hunk 1]
```diff
@@ -68,6 +68,14 @@ local function do_configure(driver, device)
   end
 end
 
+--- Handle device added
+---
+--- @param self st.zwave.Driver
+--- @param device st.zwave.Device
+local function added_handler(self, device)
+  capabilities.defaults.emit_default_events(device, self.supported_capabilities)
```
Comment (varzac) [inline drivers/SmartThings/zwave-siren/src/init.lua:76]: My only concern here is if we want the default to be to do this for _all_ the capabilities or if we want to be more specific to the problem events?  Also if any of the sub-drivers have their own `added` handlers this won't happen for them so we should review them to determine if this needs to be added to any of their `added` handlers as well.

Comment (greens) [reply drivers/SmartThings/zwave-siren/src/init.lua:76]: I feel pretty confident we don't lose anything by emitting it for everything

**Review by greens — APPROVED** [review]

**Review by greens — APPROVED** [review]

**Review by greens — CHANGES_REQUESTED** [review]
oh wait, can we add a unit test?

**Comment by ctowns** [comment]
I added a unit test, but it required some rework to the 'capabilities.defaults.emit_default_events()' function. I pushed up those changes here: https://github.ecodesamsung.com/iot-hub/scripting-engine/pull/1407

With the above changes, the 'emit_default_events()' function can complete successfully.

**Review by greens — APPROVED** [review]

**Comment by ctowns** [comment]
@lelandblue This change required a fix in the scripting engine in order to call the `emit_default_events()` function. Without the fix, the function call will cause an error. The scripting engine fix is in 45.8, but this has not been released to the general population yet. Since this is a breaking change for anyone not on the latest 45, we are waiting to merge until everyone gets the scripting engine change in 45.8.

Essentially because this causes problems for anyone without 45.8 who uses this device, we are waiting until 45.8 goes out to the general population in order to prevent issues. @greens do you know when we can expect 45.8 to be out to the general population?

**Comment by greens** [comment]
I am okay to merge this to alpha, since most folks using alpha will have 45. We can discuss on Tuesday next week whether it gets merged to 45 or not.

**Review by greens — APPROVED** [review]

**Pull request merged**